### PR TITLE
Moved AlgebraicSimplification and NopElimination passes to IE

### PR DESCRIPTION
**Pull request by GlebKazantaev**

### CODE REVIEW

File: ngraph/test/util/test_tools.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include <algorithm>
 
 #include "ngraph/ngraph.hpp"
+#include "ngraph/node.hpp"
```
Comment (ilyachur) [inline ngraph/test/util/test_tools.cpp:20]: Why do you add this header?

File: inference-engine/tests/functional/inference_engine/transformations/algebraic_simplification.cpp [hunk 2]
```diff
@@ -106,14 +73,12 @@ TEST(algebraic_simplification, multiply_negative_tests)
...
-    for (size_t i = 0; i < results.size(); i++)
-    {
+    for (size_t i = 0; i < results.size(); i++) {
         ASSERT_EQ(expected.at(i), results.at(i)->input_value(0).get_node_shared_ptr());
     }
 }
 
-TEST(algebraic_simplification, multiply_prod_negative)
-{
+TEST(algebraic_simplification, multiply_prod_negative) {
```
Comment (ilyachur) [inline inference-engine/tests/functional/inference_engine/transformations/algebraic_simplification.cpp:81]: These tests use v0 ops... Maybe we should modify them.

Comment (GlebKazantaev) [reply inference-engine/tests/functional/inference_engine/transformations/algebraic_simplification.cpp:81]: I plan to refactor this passes so I suggest to include test refactoring into this future effort.

Comment (GlebKazantaev) [reply inference-engine/tests/functional/inference_engine/transformations/algebraic_simplification.cpp:81]: BTW this tests were disabled in ngraph. So now they at least working.

**Review by ilyachur — APPROVED** [review]
In general LGTM. But we need to plan a refactoring of these tests.

**Comment by GlebKazantaev** [comment]
> In general LGTM. But we need to plan a refactoring of these tests.

ticket 37380 was created.

**Pull request merged**